lib/pbench/server/database/models/__init__.py

```diff
@@ -50,3 +51,44 @@ def process_result_value(self, value, dialect):
         if value is not None:
             value = value.replace(tzinfo=datetime.timezone.utc)
         return value
+
+
+class commonDBException(Exception):
+    """Generic base class for reporting errors ."""
+
+    def DuplicateValue(self):
+        """This method will be invoked if SQLAlchemy IntegrityError
+        throws UNIQUE constraint violation
+
+        ABSTRACT METHOD: override in subclass to perform operation.
+        """
+        raise NotImplementedError()
+
+    def NullKey(self):
+        """This method will be invoked if SQLAlchemy IntegrityError
+        throws NOT NULL constraint violation
+
+        ABSTRACT METHOD: override in subclass to perform operation.
+        """
+        raise NotImplementedError()
+
+    def _decode(self, exception: IntegrityError) -> Exception:
```

Comment:
The signature here is problematic:  I think there should be only one parameter -- the exception to be decoded...and that might be either `self` or `exception`, depending on whether this is really an instance method or a static method.

Also, since you're invoking this method outside the class, I don't think it should be named with the leading underscore prefix.